apps/emqx_dashboard/test/emqx_dashboard_https_SUITE.erl

```diff
@@ -129,69 +149,103 @@ t_compatibility_ssl_cert(_Config) ->
+                        <<"depth">> => 5
                     },
-                    max_connections => MaxConnection
+                    <<"max_connections">> => MaxConnection
                 }
             }
         }
     },
-    validate_https(Conf, MaxConnection, default_ssl_cert(), verify_none),
+    Apps = emqx_cth_suite_start(?FUNCTION_NAME, DashboardConf, Config),
+    [{apps, Apps}, {max_connection, MaxConnection} | Config];
+t_normal_ssl_cert('end', Config) ->
+    Apps = ?config(apps, Config),
+    emqx_cth_suite:stop(Apps),
     ok.
+t_normal_ssl_cert(Config) ->
+    MaxConnection = ?config(max_connection, Config),
+    validate_https(MaxConnection, default_ssl_cert(), verify_none).
 
-t_verify_cacertfile(_Config) ->
+t_verify_cacertfile(init, Config) ->
     MaxConnection = 1024,
-    DefaultSSLCert = default_ssl_cert(),
-    SSLCert = DefaultSSLCert#{cacertfile => <<"">>},
-    %% default #{verify => verify_none}
-    Conf = #{
-        dashboard => #{
-            listeners => #{
-                https => #{
-                    bind => 18084,
-                    cacertfile => <<"">>,
-                    max_connections => MaxConnection
+    DashboardConf = #{
+        <<"dashboard">> => #{
+            <<"listeners">> => #{
+                <<"https">> => #{
+                    <<"bind">> => 18084,
+                    <<"ssl_options">> => #{<<"cacertfile">> => <<"">>},
+                    <<"max_connections">> => MaxConnection
                 }
             }
         }
     },
-    validate_https(Conf, MaxConnection, SSLCert, verify_none),
+    Apps = emqx_cth_suite_start(?FUNCTION_NAME, DashboardConf, Config),
+    [{apps, Apps}, {max_connection, MaxConnection}, {dashboard_conf, DashboardConf} | Config];
+t_verify_cacertfile('end', Config) ->
+    Apps = ?config(apps, Config),
+    emqx_cth_suite:stop(Apps),
+    ok.
+t_verify_cacertfile(Config) ->
+    MaxConnection = ?config(max_connection, Config),
+    DefaultSSLCert = default_ssl_cert(),
+    SSLCert = DefaultSSLCert#{cacertfile => <<"">>},
+
+    %% validate with default #{verify => verify_none}
+    validate_https(MaxConnection, SSLCert, verify_none),
+
     %% verify_peer but cacertfile is empty
+    DashboardConf0 = ?config(dashboard_conf, Config),
     VerifyPeerConf1 = emqx_utils_maps:deep_put(
-        [dashboard, listeners, https, verify],
-        Conf,
-        verify_peer
+        [<<"dashboard">>, <<"listeners">>, <<"https">>, <<"ssl_options">>, <<"verify">>],
+        DashboardConf0,
+        <<"verify_peer">>
     ),
-    emqx_common_test_helpers:load_config(emqx_dashboard_schema, VerifyPeerConf1),
+    {ok, _} = emqx:update_config([<<"dashboard">>], maps:get(<<"dashboard">>, VerifyPeerConf1)),
+    ok = emqx_dashboard:stop_listeners(),
     ?assertMatch({error, [?NAME]}, emqx_dashboard:start_listeners()),
+
     %% verify_peer and cacertfile is ok.
     VerifyPeerConf2 = emqx_utils_maps:deep_put(
-        [dashboard, listeners, https, cacertfile],
```

Comment:
This test did not seem working as expected, `cacertfile` should be inside `ssl_options` map:  `[dashboard, listeners, https, ssl_options, cacertfile]` 🤔